Hash mismatch on CSV files stored in Git

Some tests are failing on my machine due to the new checksum feature:

```
filename = 'C:\\Project\\pywr\\tests\\models\\timeseries2.csv'
hash = 'a5c4032e2d8f5205ca99dedcfa4cd18e', algorithm = 'md5', kwargs = {}
actual_hash = '355d173e42f1bb58fca523074e570b46'

    def check_hash(filename, hash, algorithm='md5', **kwargs):
        """ Check the hash for filename using the named algorithm

         If the hashes do not match a HashMismatchError error is raised.
         """

        actual_hash = compute_hash(filename, algorithm=algorithm, **kwargs)

        if hash != actual_hash:
>           raise HashMismatchError('Hash mismatch using {} on file: "{}"'.format(algorithm, filename))
E           pywr.hashes.HashMismatchError: Hash mismatch using md5 on file: "C:\Project\pywr\tests\models\timeseries2.csv"

pywr\hashes.py:31: HashMismatchError
```

I'm guessing this is because on Windows Git is automatically converting the line endings in the CSV file to CRLF, whereas the checksum was done assuming LF. The actual data in the file is the same, but the checksum is different.

I'm not sure what the best solution here is. The current failure isn't *wrong*. We could force the CSV file to be stored in Git as binary. Or we could test the hash on a small HDF5 file instead?

Interesting that Appveyor isn't pick this up. We could change the settings on those builds to use Windows style line endings, to catch this kind of problem in the future?

https://help.github.com/articles/dealing-with-line-endings/

--- a/tests/models/.gitattributes
+++ b/tests/models/.gitattributes
@@ -1,0 +1,2 @@
+# we test the checksum of this file, so line endings must be preserved
+timeseries2.csv text eol=lf

